### DISCUSSION TITLE
Static Homepage

### DISCUSSION THREAD

**Discussion by iashyam** [cotes2020/jekyll-theme-chirpy#2099] [Q&A]
### Checklist

- [X] I have read and accepted the [contributing guidelines](https://github.com/cotes2020/jekyll-theme-chirpy/blob/master/docs/CONTRIBUTING.md).

### Is your feature request related to a problem? Please describe

I am also making an academic website using Chirpy. The main goal was to attach this web link to my emails and on my university homepage. This way, anyone who wants to know more about me can visit it. BUt it won't be a good idea for them to directly land on the blog page instead of an about page. I want something such that the defualt url `index.html` should take them to a static about page instead of a blog page and the blog page to be moved to a different page called well `blog`. 

### Describe the solution you'd like

I want something such that the defualt url `index.html` should take them to a static about page instead of a blog page and the blog page to be moved to a different page called well `blog`.

**Comment by kungfux** [comment]
I guess you can put the blog inside `blog` folder and configure Chirpy accordingly using `_config.yml`, so you can host custom `index.html` in the root that will redirect to `blog/index.html`.

```yml
# The base URL of your site
baseurl: ""
```

**Comment by kungfux** [comment]
This does not look like a bug report, but a community question. Let me move it to the discussion for you.

**Comment by SoumyaK4** [comment]
Hi Shyam @iashyam 
I'll suggest the use of subdomains
Like for blog I use `blog.soumyak4.in`
Similarly you can keep the blog in the `blog` subdomain 
And you can keep the static page to the main domain ex. `soumyak4.in` in my case, or create another subdomain named portfolio and use it as `porfolio.soumyak4.in`

Though you should have different repos for all subdomain sites.

If you don't want to pay for a domain name and keep using `username.github.io`

Then create a repo named `username.github.io` that will be the static site.
Create another repo named blog where you'll use chirpy and in the `_config.yml` set the url to `username.github.io` and baseurl to `/blog` like @kungfux suggested 
Then the blog would appear in `username.github.io/blog`


You can also enable pages in [your GitHub profile repo](https://github.com/iashyam/iashyam) and make it the static page

**Reply by iashyam** [reply]
I have a lot of articles and post written, which I have posted on an another blog [here](https://therestframe.tech/). And I have posted my CV elsewhere, and also my techinical articles are elsewhere. I want to combine them in one. So I want something like that a static homepage on my website which is like an About page, and then a blog which has all my writing.

**Reply by SoumyaK4** [reply]
@iashyam Gotcha, then I can suggest a better option to keep everything under one site.

1. Create a site from [chirpy starter](https://chirpy.cotes.page/posts/getting-started/#option-1-using-the-starter-recommended)

2. In the frontmatter of [about tab](https://github.com/cotes2020/chirpy-starter/blob/main/_tabs/about.md?plain=1) only write 
``` 
---
layout: home
--- 
```
So it becomes the default home page,
And you can add all cv info here

3. In the [index.html](https://github.com/cotes2020/chirpy-starter/blob/main/index.html) frontmatter, only write 
```
---
# add desired font awesome icon
icon: fas fa-info-circle
# if you want the blog to be right after home then order 2
order: 2
---
```

And it will look just like the normal homepage of chirpy

4. Copy paste the blog posts from other sites to the `_posts` directory and name the filenames according to chirpy

5. Set the categories in the post's frontmatter to something like `Blog` for normal blogs and `Technical` for the other posts

6. Read [chirpy posts guide](https://chirpy.cotes.page/posts/write-a-new-post/) to set the post's filenames and frontmatter correctly

**Reply by iashyam** [reply]
Thank you for your answer. I am trying to do precisely that. But the posts are not showing in any other page. Only the pinned post is visible for now. Also for some reason, the table is not rendering. I will push that side and update this comment with the link.

**Reply by SoumyaK4** [reply]
@iashyam let me do a test in a public repo, I'll update you

**Comment by SoumyaK4** [comment]
Hi @iashyam I've made a boilerplate for you...
Sorry it took some time I was traveling.
Please refer to this [demo repo](https://github.com/SoumyaK4/demo) I made
[Live demo](https://soumyak4.in/demo/)

## Changes

1. [Index.html](https://github.com/SoumyaK4/demo/blob/main/index.html)
2. The [static page](https://github.com/SoumyaK4/demo/blob/main/cv.md) you wanted
3. [Original sidebar](https://github.com/cotes2020/jekyll-theme-chirpy/blob/master/_includes/sidebar.html) vs [Modified sidebar](https://github.com/SoumyaK4/demo/blob/main/_includes/sidebar.html)
Create `_includes/sidebar.html` 
Changes at line 22 and additional lines at line 28
4. [Original Topbar](https://github.com/cotes2020/jekyll-theme-chirpy/blob/master/_includes/topbar.html) vs [Modified topbar](https://github.com/SoumyaK4/demo/blob/main/_includes/topbar.html)
Create `_includes/topbar.html`
Changes at line 11
5. New layout [for static page](https://github.com/SoumyaK4/demo/blob/main/_layouts/cv.html)
Create `_layouts/cv.html`

Still some minor changes might be necessary, if you can't figure them out let me know I'll share them whenever I get some time.
And remember, because there's code customisation involved, some things could break when you upgrade chirpy version, so keep a note of the changes you're making for future reference

**Reply by iashyam** [reply]
That works. Thanks for taking the time out and being a great helper; I really appreciate it. This was exactly how I wanted it. But the pagination is not working somehow. Everything else is working fine.

**Reply by SoumyaK4** [reply]
Thanks to your question I got a new chirpy modification to add in my list.
I'll look into the pagination tomorrow and add a comment or edit the existing solution above.
Please add a link to the public repo were you see pagination error, or add a screenshot.
Please mark that as an answer.
Have a good day @iashyam 💕
